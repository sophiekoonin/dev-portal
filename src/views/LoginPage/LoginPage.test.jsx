import React from 'react'
jest.mock('react-router-dom/Redirect')

beforeEach(() => {
  jest.resetAllMocks()
})

const getLoginPageWithContext = (
  context = { accessToken: null, hasError: false, username: null }
) => {
  jest.doMock('../../utils/AuthContext', () => ({
    Consumer: props => props.children(context)
  }))
  return require('./LoginPage').default
}

test('renders a login form correctly when there is no access token in the state', () => {
  const LoginPage = getLoginPageWithContext()
  const wrapper = mount(<LoginPage />)
  expect(wrapper).toMatchSnapshot()
})

test('redirects to dashboard when access token is not null or undefined', () => {
  const LoginPage = getLoginPageWithContext({
    accessToken: 'abc123',
    hasError: false,
    username: null
  })
  const wrapper = mount(<LoginPage />)
  expect(wrapper).toMatchSnapshot()
})
