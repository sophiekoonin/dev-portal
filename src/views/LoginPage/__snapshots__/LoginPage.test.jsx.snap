// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a login form 1`] = `
<div>
  <h1>
    Log in
  </h1>
  <div
    className="login-form"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="login-user"
      >
        Username
      </label>
      <input
        id="login-user"
        onChange={[Function]}
        required={true}
        value=""
      />
      <label
        htmlFor="login-pass"
      >
        Password
      </label>
      <input
        id="login-pass"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
      <button
        id="login-submit"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
